async-inn/Controllers/HotelsController.cs

```diff
@@ -8,9 +8,12 @@
 using async_inn.Data;
 using async_inn.Models;
 using async_inn.Services;
+using Microsoft.AspNetCore.Authorization;
 
 namespace async_inn.Controllers
 {
+
+    [Authorize(Roles = "District Manager, Property Manager, Agent")]

```

Comment:
Since a Property Manager and Agent aren't able to access any actions other than the Get routes (which you already annotated with AllowAnonymous) you're probably fine to make this just `[Authorize(Roles = "District Manager")]` and then you could go ahead and remove lines 56, 88, and 98.